File: DaggerScope/app/src/main/java/com/example/daggerscope/components/AppComponent.java
package com.example.daggerscope.components;

import com.example.daggerscope.App;
import com.example.daggerscope.modules.AppModule;
import com.example.daggerscope.scopes.AppScope;
import dagger.Component;

@AppScope
@Component(modules = {AppModule.class})
public interface AppComponent {
//    Person person();

    void inject(App app);
}

File: AndroidArchitectureComponentsDaggerSample/app/src/main/java/io/github/hurshi/androidarchitecturecomponentsdaggersample/components/AppComponent.kt
package io.github.hurshi.androidarchitecturecomponentsdaggersample.components

import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import io.github.hurshi.androidarchitecturecomponentsdaggersample.App
import io.github.hurshi.androidarchitecturecomponentsdaggersample.modules.AppModule
import io.github.hurshi.androidarchitecturecomponentsdaggersample.modules.MainActivityModule
import io.github.hurshi.androidarchitecturecomponentsdaggersample.modules.VMModule
import io.github.hurshi.androidarchitecturecomponentsdaggersample.scopes.AppScope

@AppScope
@Component(
    modules = [AndroidSupportInjectionModule::class,
        AppModule::class,
        MainActivityModule::class,
        VMModule::class]
)
abstract interface AppComponent : AndroidInjector<App> {

}